dspace-api/src/main/java/org/dspace/importer/external/arxiv/service/ArXivImportMetadataSourceServiceImpl.java

```diff
@@ -0,0 +1,360 @@
+        return retry(new CountByQueryCallable(query));
+    }
+
+
+    /**
+     * Find the number of records matching a query;
+     *
+     * @param query a query string to base the search on.
+     * @return the sum of the matching records over this import source
+     * @throws MetadataSourceException if the underlying methods throw any exception.
+     */
+    @Override
+    public int getNbRecords(Query query) throws MetadataSourceException {
+        return retry(new CountByQueryCallable(query));
+    }
+
+    /**
+     * Get a single record from the source by id
+     *
+     * @param id id of the record in ArXiv
+     * @return the first matching record
+     * @throws MetadataSourceException if the underlying methods throw any exception.
+     */
+
+    @Override
+    public ImportRecord getRecord(String id) throws MetadataSourceException {
+        List<ImportRecord> records = retry(new SearchByIdCallable(id));
+        return records == null || records.isEmpty() ? null : records.get(0);
+    }
+
+    /**
+     * Get a single record from the source.
+     *
+     * @param query a query matching a single record
+     * @return the first matching record
+     * @throws MetadataSourceException if the underlying methods throw any exception.
+     */
+    @Override
+    public ImportRecord getRecord(Query query) throws MetadataSourceException {
+        List<ImportRecord> records = retry(new SearchByIdCallable(query));
+        return records == null || records.isEmpty() ? null : records.get(0);
+    }
+
+    /**
+     * Initialize the class
+     *
+     * @throws Exception on generic exception
+     */
+    @Override
+    public void init() throws Exception {
+        Client client = ClientBuilder.newClient();
+        webTarget = client.target(baseAddress);
+    }
+
+    /**
+     * The string that identifies this import implementation. Preferable a URI
+     *
+     * @return the identifying uri
+     */
+    @Override
+    public String getImportSource() {
+        return "arxiv";
+    }
+
+    /**
+     * Expect this method will be not used and erased from the interface soon
+     */
+    @Override
+    public Collection<ImportRecord> findMatchingRecords(Item item) throws MetadataSourceException {
+        // FIXME: we need this method?
+        throw new MethodNotFoundException("This method is not implemented for ArXiv");
+    }
+
+    /**
+     * Finds records based on query object.
+     * Supports search by title and/or author
+     *
+     * @param query a query object to base the search on.
+     * @return a collection of import records.
+     * @throws MetadataSourceException if the underlying methods throw any exception.
+     */
+    @Override
+    public Collection<ImportRecord> findMatchingRecords(Query query) throws MetadataSourceException {
+        return retry(new FindMatchingRecordCallable(query));
+    }
+
+    private class CountByQueryCallable implements Callable<Integer> {
```

Comment:
Please add JavaDocs to this private Class.  All classes (private or public) need JavaDocs.